cura/LayerPolygon.py

```diff
@@ -39,10 +39,12 @@ def __init__(self, extruder: int, line_types: numpy.ndarray, data: numpy.ndarray
 
         self._extruder = extruder
         self._types = line_types
-        for i in range(len(self._types)):
-            if self._types[i] >= self.__number_of_types: # Got faulty line data from the engine.
-                Logger.log("w", "Found an unknown line type: %s", i)
-                self._types[i] = self.NoneType
+        unknown_types = np.where(self_types >= self_number_of_types)
+        if unknown_types:
+            # Got faulty line data from the engine.
+            for idx in unknown_types:
+                Logger.log("w", "Found an unknown line type at: %s", idx)
```

Comment:
Can you use the f-string and the `warn` method here?
```suggestion
                Logger.warn(f"Found an unknown line type at: {idx}")
```